src/Client/HedgedConnectionsFactory.cpp

```diff
@@ -215,7 +215,11 @@ HedgedConnectionsFactory::State HedgedConnectionsFactory::processEpollEvents(boo
                 return state;
         }
         else if (timeout_fd_to_replica_index.contains(event_fd))
```

Comment:
Not related to your PR, but it took me some effort to realize that `timeout_fd_to_replica_index` contains file descriptors of timers responsible for checking the timeout for a particular replica. The  comment `/// Map timeout for changing replica to replica index.` didn't really help, maybe something like `Maps file descriptor of timer responsible for checking replica timeout, to replica index` would be more clear.